wicketstuff-pdf-js-parent/wicketstuff-pdfjs/src/main/java/org/wicketstuff/pdfjs/res/wicketstuff-pdf.js

```diff
@@ -73,9 +89,116 @@
                         }
                     });
                     Wicket.Event.publish(WicketStuff.PDFJS.Topic.CURRENT_PAGE, pageNum, {"canvasId": config.canvasId});
+                    Wicket.Event.publish(WicketStuff.PDFJS.Topic.CURRENT_ZOOM, autoScale || scale.toFixed(2), {"canvasId": config.canvasId});
                 });
             }
 
+            function printDocument() {
+                pageRendering = true;
+                abortingPrinting = false;
+                var pages = pdfDoc.numPages;
+
+                var body = document.getElementsByTagName('body')[0];
+                body.setAttribute('pdf-js-printing', true);
+
+                var pc = document.createElement('div');
+                pc.id = 'pdf-js-print-container';
+                body.appendChild(pc);
+
+                for (var i = 0; i < pages; i++){
+                   var wrapper = document.createElement('div');
+                   pc.appendChild(wrapper);
+                }
+
+                for (var i = 1; i <= pages; i++){
+                    printPage(i, pages);
+                }
+            }
+
+            function cleanUp(error) {
+                if (abortingPrinting) { return; }
+
+                if (error) {
+                    abortingPrinting = true;
+                    window.alert("error printing!");
```

Comment:
Absolutely it would. I need to add progress and success/fail notifications as it can take quite a while to render each page for printing. The alert is just temporary until I've added those early next week when I get back to this.